api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
+from ..people.models import Role, Person
+from src.shared.helpers import modify_entity, get_all_queried_entities
 
-# ---- Group
+from src.shared.models import QueryArgumentError
 
-group_schema = GroupSchema()
+from sqlalchemy.exc import IntegrityError, DBAPIError
 
+# ---- Group Type
 
-def group_dump(group):
-    group.manager_info = group.manager
-    group.manager_info.person = group.manager.person
-    group.member_list = group.members
-    return jsonify(group_schema.dump(group))
+group_type_schema = GroupTypeSchema()
 
+@groups.route('/group-types', methods=['POST'])
+def create_group_type():
+    try:
+        valid_group_type = group_type_schema.load(request.json)
+    except ValidationError as err:
+        return jsonify(err.messages), 422
 
-@groups.route('/groups', methods=['POST'])
+    new_group_type = GroupType(**valid_group_type)
+    db.session.add(new_group_type)
+    db.session.commit()
+    return jsonify(group_type_schema.dump(new_group_type)), 201
+
+@groups.route('/group-types/<int:group_type_id>', methods=['GET'])
+def read_one_group_type(group_type_id):
+    group_type = db.session.query(GroupType).filter_by(id=group_type_id).first()
+    if not group_type:
+        return jsonify(f"GroupType with id #{group_type_id} does not exist."), 404
+
+    return jsonify(group_type_schema.dump(group_type))
+
+@groups.route('/group-types', methods=['GET'])
+def read_all_group_types():
+    query = db.session.query(GroupType)
+    try:
+        group_types = get_all_queried_entities(query, request.args)
+    except QueryArgumentError as e:
+        return jsonify(e.message), e.code
+    return jsonify(group_type_schema.dump(group_types, many=True))
+
+@groups.route('/group-types/<int:group_type_id>', methods=['PATCH'])
 @jwt_required
-def create_group():
-    request.json['active'] = True
-    members_to_add = None
-    if 'person_ids' in request.json.keys():
-        members_to_add = request.json['person_ids']
-        del request.json['person_ids']
+def update_group_type(group_type_id):
+    group_type_schema = GroupTypeSchema()
 
     try:
-        valid_group = group_schema.load(request.json)
+        valid_attributes = group_type_schema.load(request.json, partial=True)
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    new_group = Group(**valid_group)
+    return modify_entity(GroupType, group_type_schema, group_type_id, valid_attributes)
 
-    if db.session.query(Manager).filter_by(id=new_group.manager_id).first() is None:
-        return jsonify(msg="Manager not found"), 404
 
-    db.session.add(new_group)
+@groups.route('/group-types/<int:group_type_id>', methods=['DELETE'])
+@jwt_required
+def delete_group_type(group_type_id):
+    group_type = db.session.query(GroupType).filter_by(id=group_type_id).first()
 
-    today = datetime.datetime.today().strftime('%Y-%m-%d')
+    if not group_type:
+        return jsonify(f"GroupType with id #{group_type_id} does not exist."), 404
```

Comment:
Here's another instance of that `jsonify(... text ...)` thing.

Reply:
yes..